jujugui/static/gui/src/app/components/user-profile/budget-list.js

```diff
@@ -0,0 +1,184 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2016 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('budget-list', function() {
+
+  juju.components.BudgetList = React.createClass({
+    propTypes: {
+      listBudgets: React.PropTypes.func.isRequired,
+      user: React.PropTypes.object,
+    },
+
+    getInitialState: function() {
+      this.xhrs = [];
+
+      return {
+        budgetList: [],
+        loadingBudgets: false,
+      };
+    },
+
+    componentWillMount: function() {
+      this._getBudgets();
+    },
+
+    componentWillUnmount: function() {
+      this.xhrs.forEach((xhr) => {
+        xhr && xhr.abort && xhr.abort();
+      });
+    },
+
+    componentWillReceiveProps: function(nextProps) {
+      // If the user has changed then update the data.
+      var props = this.props;
+      var currentUser = props.user && props.user.user;
+      var nextUser = nextProps.user && nextProps.user.user;
+      if (nextUser !== currentUser) {
+        this._getBudgets();
+      }
```

Comment:
I'll add a test in a follow-up card, since this PR is primarily to split up existing code without adding much new functionality.